core/ops.rs

```diff
@@ -201,7 +201,16 @@ impl OpTable {
     F: Fn(Rc<RefCell<OpState>>, OpPayload) -> Op + 'static,
   {
     let (op_id, prev) = self.0.insert_full(name.to_owned(), Rc::new(op_fn));
-    assert!(prev.is_none());
+    assert!(prev.is_none(), "Op '{}' was already registered", name);
+    op_id
+  }
+
+  pub fn replace_op<F>(&mut self, name: &str, op_fn: F) -> OpId
+  where
+    F: Fn(Rc<RefCell<OpState>>, OpPayload) -> Op + 'static,
+  {
```

Comment:
Side discussion: it might makes sense to someday rewrite these (replace_op, register_op, etc.) to:
```rust
pub fn frob_op(&mut self, name: &str, op_fn: Rc<OpFn>) {
  // ...
}
```
Because right now they turn into a combinatorial explosion of monomorphized calls.

Truthfully, that's probably more relevant for `op_sync()` and `op_async()` because they're called way more often than `register_op()` - approx. 300 vs. 35 calls.

Reply:
Mid-term I would be in favour of only allowing ops to be registered declaratively at init via Extensions.